### PR TITLE
Accounts and Test Suites in GitHub Action

### PR DESCRIPTION
**Pull request by iainjwhiteside**
Adds accounts and test suites to the github action

### CODE REVIEW

**Comment by iainjwhiteside** [comment]
still need to create the test suite so the CI passes

File: __tests__/main.test.ts [hunk 1]
```diff
@@ -157,4 +219,75 @@ describe('action', () => {
...
+    const testSuiteBatchInputMock = jest
+      .spyOn(BatchesApi.prototype, 'createBatchForTestSuite')
+      .mockResolvedValueOnce({ data: newBatch } as AxiosResponse)
+
+    await main.run()
+
+    expect(getProjectIDMock).toHaveBeenCalledTimes(1)
+    expect(getSystemIDMock).toHaveBeenCalledTimes(1)
+    expect(getTestSuiteIDMock).toHaveBeenCalledTimes(1)
+    expect(createBranchMock).not.toHaveBeenCalled()
```
Comment (benabineri) [inline __tests__/main.test.ts:288]: Once?

File: src/main.ts [hunk 2]
```diff
@@ -106,39 +129,80 @@ export async function run(): Promise<void> {
...
+    if (core.getInput('test_suite') !== '') {
+      const testSuiteName = core.getInput('test_suite')
+      // Create a test suite batch input
+      const runSuiteRequest: TestSuiteBatchInput = {
+        buildID: newBuild.buildID
+      }
+      // Check that newBuild.buildID is defined and then use its string:
+      if (newBuild.buildID !== undefined) {
+        runSuiteRequest.buildID = newBuild.buildID
+      }
```
Comment (benabineri) [inline src/main.ts:151]: Isn't there duplication here?

File: src/test_suites.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+export async function getTestSuiteID(
+  batchesApi: BatchesApi,
+  projectID: string,
+  testSuiteName: string
+): Promise<string> {
+  const suites = await listTestSuites(projectID, batchesApi)
+  const thisTestSuite = suites.find(ts => ts.name === testSuiteName)
+  if (thisTestSuite?.systemID !== undefined) {
+    return thisTestSuite.systemID
```
[the remark this answers is not in the document]
Comment (iainjwhiteside) [reply src/test_suites.ts:31]: feck. copy+paste fud

[the remark this answers is not in the document]
Comment (iainjwhiteside) [reply src/test_suites.ts:31]: the e2e test will definitely have caught that, not the mocks, of course

File: __tests__/main.test.ts (hunk 1 above)
Comment (iainjwhiteside) [reply __tests__/main.test.ts:288]: this one is not called, but the batch is one time too, i can add that

**Review by benabineri — APPROVED** [review]

**Pull request merged**